GRC example: loopback-codec2 fails

### What happened?

Checking upon https://github.com/gnuradio/gnuradio/pull/4208 , I tried running 

    python3 -m pytest ../grc/tests/test_examples.py  

from my build directory. All fine, but for:

```text
example = <DirEntry 'loopback-codec2.grc'>
[…]
>       assert flow_graph.is_valid(), (example.name, [
            f"{print_proper(elem)}: {msg}" for elem, msg in flow_graph.iter_error_messages()])

../grc/tests/test_examples.py:95:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../grc/tests/test_examples.py:95: in <listcomp>
    assert flow_graph.is_valid(), (example.name, [
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

element = <grc.core.Connection.Connection object at 0x7f6a9d1a4bd0>

    def print_proper(element):
        if element.is_block:
            return element.name
>       return f"{element.parent.name} - {element}"
E       AttributeError: 'FlowGraph' object has no attribute 'name'
```

### System Information

OS: F37 x86_64
GR Installation Method: src build


### GNU Radio Version

4.0-git (dev-4.0)

### Specific Version

d5478261570265122a5aea7f5fbf08808f0d3418

### Steps to Reproduce the Problem

```shell
mkdir build && cd build
cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=$GRPREFIX -GNinja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER=clang++  -DCMAKE_C_COMPILER=clang -DENABLE_CTRLPORT_THRIFT=Off  -Wno-dev  -DENABLE_GR_CTRLPORT=ON -DENABLE_GNURADIO_RUNTIME=ON -DENABLE_PERFORMANCE_COUNTERS=Off -DCMAKE_CXX_FLAGS=-D_FORTIFY_SOURCE=3 ..
ninja
ninja install > /dev/null
python3 -m pytest ../grc/tests/test_examples.py
```

### Relevant log output

```shell
example = <DirEntry 'loopback-codec2.grc'>

    @pytest.mark.examples
    @pytest.mark.parametrize("example", gather_examples())
    def test_all_examples(example):
        global BLOCK_PATHS

        print(example.path)

        platform = Platform(
            name='GNU Radio Companion Compiler',
            prefs=None,
            version='0.0.0',
        )
        platform.build_library(BLOCK_PATHS)

        flow_graph = platform.make_flow_graph(example.path)
        flow_graph.rewrite()
        flow_graph.validate()

>       assert flow_graph.is_valid(), (example.name, [
            f"{print_proper(elem)}: {msg}" for elem, msg in flow_graph.iter_error_messages()])

../grc/tests/test_examples.py:95:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../grc/tests/test_examples.py:95: in <listcomp>
    assert flow_graph.is_valid(), (example.name, [
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

element = <grc.core.Connection.Connection object at 0x7f6a9d1a4bd0>

    def print_proper(element):
        if element.is_block:
            return element.name
>       return f"{element.parent.name} - {element}"
E       AttributeError: 'FlowGraph' object has no attribute 'name'

../grc/tests/test_examples.py:74: AttributeError
--------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------
/media/stripedvolume/src/gnuradio/grc/tests/../../gr-vocoder/examples/loopback-codec2.grc
=========================================================================================================== warnings summary ===========================================================================================================
../grc/tests/test_examples.py:77
  /media/stripedvolume/src/gnuradio/grc/tests/test_examples.py:77: PytestUnknownMarkWarning: Unknown pytest.mark.examples - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.examples

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================================================= short test summary info ========================================================================================================
FAILED ../grc/tests/test_examples.py::test_all_examples[example20] - AttributeError: 'FlowGraph' object has no attribute 'name'
========================================================================================= 1 failed, 214 passed, 1 warning in 81.78s (0:01:21) ==========================================================================================
```

--- a/gr-vocoder/grc/vocoder_codec2_decode_ps.block.yml
+++ b/gr-vocoder/grc/vocoder_codec2_decode_ps.block.yml
@@ -21,7 +21,7 @@ parameters:
 inputs:
 -   domain: stream
     dtype: byte
-    vlen: '${ { 0: 64, 1: 48, 2: 64, 3: 56, 4: 52, 5: 48, 6: 28, 7: 28, 8: 28, 9: 64, 10: 18, 11: 18 }[mode] }'
+    vlen: '${ {"codec2.MODE_3200": 64, "codec2.MODE_2400": 48, "codec2.MODE_1600": 64, "codec2.MODE_1400": 56, "codec2.MODE_1300": 52, "codec2.MODE_1200": 48, "codec2.MODE_700": 28, "codec2.MODE_700B": 28, "codec2.MODE_700C": 28, "codec2.MODE_WB": 64, "codec2.MODE_450": 18, "codec2.MODE_450PWB": 18}[mode] }'
 
 outputs:
 -   domain: stream

--- a/gr-vocoder/grc/vocoder_codec2_encode_sp.block.yml
+++ b/gr-vocoder/grc/vocoder_codec2_encode_sp.block.yml
@@ -25,7 +25,7 @@ inputs:
 outputs:
 -   domain: stream
     dtype: byte
-    vlen: '${ { 0: 64, 1: 48, 2: 64, 3: 56, 4: 52, 5: 48, 6: 28, 7: 28, 8: 28, 9: 64, 10: 18, 11: 18 }[mode] }'
+    vlen: '${ {"codec2.MODE_3200": 64, "codec2.MODE_2400": 48, "codec2.MODE_1600": 64, "codec2.MODE_1400": 56, "codec2.MODE_1300": 52, "codec2.MODE_1200": 48, "codec2.MODE_700": 28, "codec2.MODE_700B": 28, "codec2.MODE_700C": 28, "codec2.MODE_WB": 64, "codec2.MODE_450": 18, "codec2.MODE_450PWB": 18}[mode] }'
 
 templates:
     imports: |-

--- a/grc/core/utils/descriptors/evaluated.py
+++ b/grc/core/utils/descriptors/evaluated.py
@@ -23,8 +23,7 @@ class Evaluated(object):
             value = instance.parent_block.evaluate(raw)
         except Exception as error:
             if raw:
-                instance.add_error_message(
-                    "Failed to eval '{}': {}".format(raw, error))
+                instance.add_error_message(f"Failed to eval '{raw}': ({type(error)}) {error}")
             return self.default
 
         if not isinstance(value, self.expected_type):

